statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,626 @@
+        if cov_method not in ["boot", 'mean_boot', 'jack']:
+            raise ValueError("Method %s not supported" % cov_method)
+
+        if self.rep_weights is not None:
+            # should rep_weights be a list of arrays or a ndarray
+            return self.rep_weights[:, c]
+        if cov_method == 'jack':
+            return self._jackknife_rep_weights(c)
+        elif cov_method == 'boot':
+            return self._bootstrap_weights()
+        else:
+            return self._mean_bootstrap_weight(bsn)
+
+    def _jackknife_rep_weights(self, c):
+        """
+        Computes 'delete 1' jackknife replicate weights
+
+        Parameters
+        ----------
+        c : integer or None
+            Represents which cluster to leave out when computing
+            'delete 1' jackknife replicate weights
+
+        Returns
+        -------
+        w : ndarray
+            Augmented weight
+        """
+        # get stratum that the cluster belongs in
+        s = self.strat_for_clust[c]
+        nh = self.clust_per_strat[s]
+        w = self.weights.copy()
+        # all weights within the stratum are modified
+        w[self.strat == s] *= nh / float(nh - 1)
+        # but if you're within the cluster to be removed, set as 0
+        w[self.sclust == c] = 0
+        return w
+
+    def _bootstrap_weights(self):
+        """
+        Computes bootstrap replicate weight
+
+        Returns
+        -------
+        w : ndarray
+            Augmented weight
+        """
+        w = self.weights.copy()
+        clust_count = np.zeros(self.nclust)
+        for s in range(self.n_strat):
+            # how to handle strata w/ only one cluster?
+            w[self.strat == s] *= float(self.clust_per_strat[s] - 1) \
+                                         / self.clust_per_strat[s]
+            # If there is only one cluster then weights wont change
+            if self.clust_per_strat[s] == 1:
+                continue
+
+            # resample array of clusters
+            ii_resample = np.random.choice(self.ii[s], size=(self.clust_per_strat[s]-1))
+            # accumulate number of times cluster i was resampled
+            clust_count += np.bincount(ii_resample,
+                                       minlength=max(self.sclust)+1)
+
+        w *= clust_count[self.sclust]
+        return w
+
+    def _mean_bootstrap_weight(self, bsn):
+        """
+        Computes mean bootstrap replicate weight
+
+        Parameters
+        ----------
+        bsn : integer
+            Mean bootstrap averages the number of resampled clusters over bsn
+        Returns
+        -------
+        w : ndarray
+            Augmented weight
+        """
+        clust_count = np.zeros(self.design.nclust)
+        w = self.weights.copy()
+        # for each replicate, I accumulate bsn number of times?
+        for b in range(bsn):
+            for s in range(self.n_strat):
+                w[self.strat == s] *= ((float(self.clust_per_strat[s] - 1) /
```

Comment:
We still support Python2 so `1/bsn` may be a problem (1/2 = 0 in Python2 if 2 has integer type).  I don't think you need the first float here, but you should add one around 1/bsn.